### ISSUE TITLE
508-defect-3 [AXE-CORE]: Elements MUST have their visible text as part of their accessible name - Feedback button issue

### ISSUE THREAD

**Issue by jenstrickland** [department-of-veterans-affairs/va.gov-team#6377]
**Feedback framework**

- **❗️ Must** for if the feedback must be applied 
- **⚠️Should** if the feedback is best practice 
- **✔️ Consider** for suggestions/enhancements 

## Description

 Elements **MUST** have the accessible name — such as the aria-label — contain the text of the visible label in the same order in which it appears. While it is not necessary that the visible label and accessible name match exactly, it is a best practice to have the accessible name begin with the visible text. In this case, the text inside the element is currently not included in the accessible name.

### Why it Matters

Speech input users can interact with a webpage by speaking the visible text labels of menus, links, and buttons that appear on the screen. It is confusing to speech input users when they say a visible text label they see, but the speech command does not work because the component's accessible (programmatic) name does not match the visible label. 

When a user interface component has a visible text label — whether it be real text or an image of text — that text must also be found in the component's accessible (programmatic) name. When the visible label and accessible (programmatic) name for interactive components are in sync, speech input users can effectively interact with those components.

This rule applies to any element that has:

- a semantic role that is a widget that supports name from content, and
- visible text content, and
- an aria-label or aria-labelledby attribute.

**Note:** widget roles that supports name from content are: button, checkbox, gridcell, link, menuitem, menuitemcheckbox, menuitemradio, option, radio, searchbox, switch, tab, treeitem.

**Expectation:** The complete visible text content of the target element either matches or is contained within its accessible name.

**Note:** Leading and trailing whitespace and difference in case sensitivity should be ignored.

## Point of Contact

**VFS Point of Contact:** Jennifer

## Acceptance Criteria

As a speech input user, I want to interact with the map controls by speaking the visible text labels.

## Environment

* Operating System: any
* Browser: any
* Server destination: production

## Steps to Recreate

1. Enter `https://www.va.gov/decision-reviews/` in browser (or pretty much anywhere on VA.gov that displays the Feedback button
2. Scan the page using axe-coconut extension (https://chrome.google.com/webstore/detail/axe-coconut-web-accessibi/iobddmbdndbbbfjopjdgadphaoihpojp?hl=en)
3. Verify there is an issue with the text and name/label
4. As a speech input user, test that activating the button works using the visible text of the button

## WCAG or Vendor Guidance (optional)

* [Label and name from content mismatch](https://dequeuniversity.com/rules/axe/3.5/label-content-name-mismatch?application=coconut)
* [W3C aria-haspopup (property)](https://www.w3.org/WAI/PF/aria/states_and_properties#aria-haspopup)

## Screenshots or Trace Logs

![VAgov-Feedback-visible-text-name](https://user-images.githubusercontent.com/57469/75718014-3b70ea80-5ca0-11ea-9d51-13a0d3a23d8b.png)

**Comment by meganhkelley** [comment]
@caw310 @cvalarida I'm adding y'all to this and removing our label under the assumption that this is a design system defect — if not, please add my label back and @ me! Thanks.

**Comment by k80bowman** [comment]
@jwoodman5 is your team responsible for the Medallia button? I think this issue may be resolved, but before I close it I wanted to check with you or your team and make sure.

**Comment by jwoodman5** [comment]
@aubreyarcangel Are you able to confirm this is resolved with the new buttton?

**Comment by caw310** [comment]
@k80bowman checked and this issue is resolved.
